### PR TITLE
CI: Run tests on swiftserver

### PR DESCRIPTION
**Pull request by adam-fowler**
Run `npm test`
Requires it to be run by `xvfb` in a separate shell file. If you include in docker-compose file it doesn't run
Had to install a bunch of additional stuff in Dockerfile to get it to work
Also added fix for test that only works in swift 5.6

### CODE REVIEW

**Comment by adam-fowler** [comment]
@tomerd if you are happy with this can you enable the swift 5.6 and 5.4 targets.

**Comment by tomerd** [comment]
@swift-server-bot test this please

File: docker/test.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+xvfb-run -a npm test
```
Comment (tomerd) [inline docker/test.sh:1]: probably belongs in `script` rather than `docker` directory, but up to you

Comment (adam-fowler) [reply docker/test.sh:1]: I'm happy with it where it is. It's a single line script which is only ever called by the docker-compose file

**Review by 0xTim — APPROVED** [review]

**Pull request merged**